ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/common/FailureXAResource.java

```diff
@@ -9,157 +9,131 @@
+                    throw new XAException(XAException.XAER_NOTA);
+                case inval:
+                    throw new XAException(XAException.XAER_INVAL);
+                case proto:
+                    throw new XAException(XAException.XAER_PROTO);
+                case rmfail:
+                    throw new XAException(XAException.XAER_RMFAIL);
+                case XA_RBINTEGRITY:
+                    throw new XAException(XAException.XA_RBINTEGRITY);
+                default:
+                    throw new XAException(XAException.XA_RBTIMEOUT);
+            }
+        }
+        // successful
+        this._committed = true;
     }
 
-    public void end(Xid xid, int flags) throws XAException
-    {
-        if (_locale == FailLocation.end)
-        {
-            if (_type == FailType.normal)
-                throw new XAException(XAException.XA_HEURRB);
-            
-            if (_type == FailType.timeout)
-                throw new XAException(XAException.XA_RBTIMEOUT);
-            
-            if (_type == FailType.XA_RBCOMMFAIL)
-                throw new XAException(XAException.XA_RBCOMMFAIL);
+    public void end(Xid xid, int flags) throws XAException {
+        if (_locale == FailLocation.end) {
+            switch (_type) {
+                case normal:
+                    throw new XAException(XAException.XA_HEURRB);
+                case timeout:
+                    throw new XAException(XAException.XA_RBTIMEOUT);
+                case XA_RBCOMMFAIL:
+                    throw new XAException(XAException.XA_RBCOMMFAIL);
+            }
         }
     }
 
-    public void forget(Xid xid) throws XAException
-    {
+    public void forget(Xid xid) throws XAException {
     }
 
-    public int getTransactionTimeout() throws XAException
-    {
-	return 0;
+    public int getTransactionTimeout() throws XAException {
+        return 0;
     }
 
-    public boolean isSameRM(XAResource xares) throws XAException
-    {
-	return false;
+    public boolean isSameRM(XAResource xares) throws XAException {
+        return false;
     }
 
-    public int prepare(Xid xid) throws XAException
-    {
-        if ((_locale == FailLocation.prepare) || (_locale == FailLocation.prepare_and_rollback))
-        {
-            if (_type == FailType.message) {
-                XAException xae = new XAException(XAException.XA_RBROLLBACK);
-                xae.initCause(new Throwable("test message"));
-                throw xae;
-            } else if(_type == FailType.XA_HEURHAZ) {// XA spec invalid error code
-                throw new XAException(XAException.XA_HEURHAZ);
-            } else if(_type == FailType.XA_RBINTEGRITY) {
-                throw new XAException(XAException.XA_RBINTEGRITY);
-            } else {
-                throw new XAException(XAException.XAER_INVAL);
+    public int prepare(Xid xid) throws XAException {
+        if ((_locale == FailLocation.prepare) || (_locale == FailLocation.prepare_and_rollback)) {
```

Comment:
Noting here 3/3 changes from if -> switch

Reply:
I rolled back this change.